src/pymovements/gaze/transforms.py

```diff
@@ -803,13 +810,31 @@ def resample(
     # Sort columns by datetime
     samples = samples.sort('datetime')
 
-    # Replace pre-existing null values with NaN as they should not be interpolated
+    # Helper: determine numeric columns among the target columns
+    def _is_numeric_dtype(dtype: pl.DataType) -> bool:
+        base_dtype = dtype.inner if isinstance(dtype, pl.List) else dtype
+        return (
+            base_dtype in {
+                pl.Int8, pl.Int16, pl.Int32, pl.Int64,
+                pl.UInt8, pl.UInt16, pl.UInt32, pl.UInt64,
+                pl.Float32, pl.Float64,
```

Comment:
all of these types are subclasses of [`pl.NumericType`](https://github.com/pola-rs/polars/blob/ca54d5c8ba19264f8ea2cc3ee36536895fa095e1/py-polars/src/polars/datatypes/classes.py#L286). It would be more future-proof to just check is `issubclass(base_dtype, pl.NumericType)`.